### PR TITLE
HDDS-8005. Fixed intermittent failure in TestOmSnapshot.testSnapDiffWithMultipleSSTs

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
`TestOmSnapshot.testSnapDiffWithMultipleSSTs` was failing all the time because snapshot diff is an async API and when a job is submitted for the first time, it always returns empty diff list with IN_PROGRESS status. User is supposed to make second call to fetch the results, once job is DONE. TestOmSnapshot.testSnapDiffWithMultipleSSTs was only making one call which only submits the request and returned empty diff response as mentioned above.
This change is to call snap diff after job to DONE to fetch complete diff report.

## What is the link to the Apache JIRA
HDDS-8005

## How was this patch tested?
* Ran the test locally and on fork: https://github.com/hemantk-12/ozone/actions/runs/7620429542
* Flaky-test-check workflow run: https://github.com/hemantk-12/ozone/actions/runs/7630045775

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOmSnapshot.java [hunk 1]
```diff
@@ -1800,29 +1797,27 @@ public void testSnapDiffWithMultipleSSTs()
     String keyPrefix = "key-";
     // add file to bucket1 and take snapshot
     createFileKeyWithPrefix(bucket1, keyPrefix);
+    int keyTableSize = getKeyTableSstFiles().size();
```
Comment (sadanand48) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOmSnapshot.java:1800]: I think keyTableSize will always be 0. Why is this added?

**Comment by sadanand48** [comment]
Thanks @hemantk-12 , overall change looks good , left a minor comment. Could you also please run the flaky test workflow for this test a few times.

**Review by aswinshakil — APPROVED** [review]
Thanks for the patch. We can just run the`falky-test-check` just to be sure. Otherwise I'm good with the changes.

**Comment by hemantk-12** [comment]
Initially, I didn't run the `flaky-test-check` because this test was marked `Unhealthy` not `Flaky` which means it failed all the times.

Just to be sure as suggested above, I ran it for 5x10 for one of the TestOmSnapshot's implementation: https://github.com/hemantk-12/ozone/actions/runs/7630045775 and it passed all the times.

**Comment by hemantk-12** [comment]
Thanks @sadanand48 and @aswinshakil for the review.
